neuro_san/http_sidecar/http_sidecar.py

```diff
@@ -68,46 +70,39 @@ def __call__(self):
         app.listen(self.http_port)
         self.logger.info({}, "HTTP server is running on port %d", self.http_port)
         self.logger.debug({}, "Serving agents: %s", repr(self.agents.keys()))
+        # Put agent names into "allowed" list:
+        for agent_name, _ in self.agents.items():
+            self.allowed_agents[agent_name] = True
         IOLoop.current().start()
 
     def make_app(self):
         """
         Construct tornado HTTP "application" to run.
         """
+        request_data: Dict[str, Any] = self.build_request_data()
         handlers = []
         handlers.append(("/", HealthCheckHandler))
-        concierge_data: Dict[str, Any] = self.build_request_data("concierge")
-        handlers.append(("/api/v1/list", ConciergeHandler, concierge_data))
-        openapi_spec_data: Dict[str, Any] = self.build_request_data("openapi")
-        handlers.append(("/api/v1/docs", OpenApiPublishHandler, openapi_spec_data))
+        handlers.append(("/api/v1/list", ConciergeHandler, request_data))
+        handlers.append(("/api/v1/docs", OpenApiPublishHandler, request_data))
 
-        for agent_name in self.agents.keys():
-            # For each of registered agents, we define 3 request paths -
-            # one for each of neuro-san service API methods.
-            # For each request http path, we build corresponding request handler
-            # and put it in "handlers" list,
-            # which is used to construct tornado "application".
-            request_data: Dict[str, Any] = self.build_request_data(agent_name)
-            route: str = f"/api/v1/{agent_name}/connectivity"
-            handlers.append((route, ConnectivityHandler, request_data))
-            self.logger.info({}, "Registering URL path: %s", route)
-            route: str = f"/api/v1/{agent_name}/function"
-            handlers.append((route, FunctionHandler, request_data))
-            self.logger.info({}, "Registering URL path: %s", route)
-            route: str = f"/api/v1/{agent_name}/streaming_chat"
-            handlers.append((route, StreamingChatHandler, request_data))
-            self.logger.info({}, "Registering URL path: %s", route)
+        # Register templated request paths for agent API methods:
+        handlers.append((r"/api/v1/([^/]+)/function", FunctionHandler, request_data))
+        handlers.append((r"/api/v1/([^/]+)/connectivity", ConnectivityHandler, request_data))
+        handlers.append((r"/api/v1/([^/]+)/streaming_chat", StreamingChatHandler, request_data))
 
         return Application(handlers)
 
-    def build_request_data(self, agent_name: str) -> Dict[str, Any]:
+    def allow(self, agent_name) -> bool:
+        return self.allowed_agents.get(agent_name, False)
+
```

Comment:
This method is our policy implementation - agent name is allowed for incoming requests,
if it is in self.allowed_agents dictionary and mapped to True.